### PR TITLE
feat(uninstall): add flag allow-not-found

### LINKED ISSUE #9687: helm uninstall: erroneous exit code if no release found
If a release was not previously installed and `helm uninstall release-name` is run the following error occurs: 

```bash
Error: uninstall: Release not loaded: release-name: release: not found
```
That's fine and having read your design considerations I understand the reasoning behind the default behavior and it should not be altered. 

Looking at it from the automation perspective I would propose it as useful to add a flag similar to the `helm upgrade --install` flag. It would accept `release: not found` as successful uninstallation, as technically it was successfully uninstalled. 
Currently running the command in a CI pipeline errors if e.g. a previous installation wasn't successful and everything is teared down. 

Currently one has to use `|| true` but this has the drawback that it hides all errors, even those that might really happen during an uninstall. 

I have searched for similar issues but haven't found one, sorry if I might have overlooked it.

Output of `helm version`: `version.BuildInfo{Version:"v3.5.3", GitCommit:"041ce5a2c17a58be0fcd5f5e16fb3e7e95fea622", GitTreeState:"dirty", GoVersion:"go1.15.8"}`

### PR DESCRIPTION
**Pull request by cndoit18**
**What this PR does / why we need it**:

closes #9687 

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by cndoit18** [comment]
Hello, this is very simple, it may be useful in the CI / CD, can help review it? please
@bacongobbler @mattfarina

File: pkg/action/uninstall_test.go [hunk 1]
```diff
@@ -60,3 +60,16 @@ func TestUninstallRelease_deleteRelease(t *testing.T) {
...
+
+func TestUninstallRelease_allowNotFound(t *testing.T) {
+	is := assert.New(t)
+
+	unAction := uninstallAction(t)
+	unAction.DisableHooks = true
+	unAction.DryRun = false
+	unAction.KeepHistory = true
+	unAction.AllowNotFound = true
+	res, err := unAction.Run("keep-secret")
```
Comment (baijum) [inline pkg/action/uninstall_test.go:72]:
```suggestion
	res, err := unAction.Run("chart-release-that-does-not-exist")
```
How about a more descriptive name?

Comment (cndoit18) [reply pkg/action/uninstall_test.go]: great

**Comment by kladiv** [comment]
+1 really useful

**Comment by kavishdahekar** [comment]
+1 we'd benefit from this.
Can somebody merge this PR. It's open since 2021 😅 .

**Comment by vakenbolt** [comment]
Please merge this! 🙏

**Comment by cndoit18** [comment]
same pr: https://github.com/helm/helm/pull/10564

**Pull request closed without merging**